src/ExtendedXmlSerializer/ExtendedXmlSerializerConfig.cs

```diff
@@ -132,6 +134,11 @@ void IExtendedXmlSerializerConfig.WriteObject(XmlWriter writer, object obj)
             _serializer(writer, (T) obj);
         }
 
+        public bool IsSatisfiedBy( Type type )
+        {
+            return specification( type );
```

Comment:
I am partial to delegates as it allows for more flexibility (and therefore power).  An extending consumer can put ANYTHING in there and it will be executed to their liking.  This works best if there is a default implementation, which of course there is.

So really, this is about passing power onto those that use/extend your API.  The more power you (reasonably 😎) give, the more it is like/adopted.